index.bs

```diff
@@ -357,10 +355,46 @@ execute the following steps:
 5.  [$Set the `Sec-CH-UA` header for a request$], given |r|.
 
 
-IDL {#idl} 
-===
+Interface {#interface} 
+=================
+
+<pre class="idl">
+[Exposed=Window]
+interface NavigatorUAData {
+  readonly attribute DOMString brand;
+  readonly attribute DOMString version;
+  readonly attribute DOMString platform;
+  readonly attribute DOMString architecture;
+  readonly attribute DOMString model;
+  readonly attribute boolean mobile;
+};
+interface mixin NavigatorUA {
+  [SecureContext] Promise&lt;NavigatorUAData&gt; getUserAgent();
+};
+Navigator includes NavigatorUA;
+
+</pre>
 
-ISSUE: Yoav should write all this.
+Processing model {#processing}
+--------------
+<dfn method for="NavigatorUA"><code>getUserAgent()</code></dfn> method *must*
+run these steps:
+
+1. Let |UAData| be a new {{NavigatorUAData}} object.
+2. Set |UAData|'s {{NavigatorUAData/brand}} to the value of the user agent's [=user agent/brand=].
+3. Set |UAData|'s {{NavigatorUAData/platform}} to the value of the user agent's
+    [=user agent/platform brand and version=].
+4. Set |UAData|'s {{NavigatorUAData/architecture}} to the value of the user
+    agent's [=user agent/platform architecture=].
+5. Set |UAData|'s {{NavigatorUAData/model}} to the value of the user agent's [=user agent/model=].
+6. Set |UAData|'s {{NavigatorUAData/mobile}} to the value of the user agent's
+    [=user agent/mobileness=].
+7. Let |version| be the value of the user agent's [=user agent/major version=].
+8. Let |settings| be the [=context object=]’s [=relevant settings object=].
+9. If |settings|'s [=environment settings object/client-hints set=] [=list/contains=] `UA`, then
```

Comment:
Why do you check the client-hints set list here, and nowhere else? Shouldn't we check all of them or none of them?

Reply:
There's no need to check the client-hints set for the JS variables, is there?
Is there a way to get the only the major version in JS? (to avoid touching too many bits)

Reply:
I'm wondering why we're checking the client-hints set at all. If we're exposing platform and model and etc. without checking, why are we not exposing the full version here? Is your goal to ensure that developers can use less of their privacy budget by only asking for the major version?

Reply:
Removed the check and added an issue about adding an extra method for major version

Reply:
@mikewest - and answering your question that I just now saw after replying (Thanks GitHub!), we don't really need this check, and I got myself confused with the header behavior. At the same time, might be a good idea to allow checking just the major version "for free". Opened an issue